sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample07_MockEventProcessorHandlersTests.cs

```diff
@@ -0,0 +1,244 @@
+            var eventArgs = new PartitionInitializingEventArgs("0", defaultStartingPosition, cancellationSource.Token);
+
+            // Execute the handler and validate that cancellation was respected.
+
+            await initializeEventHandler(eventArgs);
+            Assert.That(eventArgs.DefaultStartingPosition, Is.EqualTo(defaultStartingPosition));
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task CloseEventHandlerOwnershipLostReason()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_CloseEventHandlerOwnershipLostReason
+
+            // Partition close event handler responsible for the detection and processing
+            // of the lost ownership reason example scenario.
+
+            bool ownershipLostDetected = false;
+
+            Task closeEventHandler(PartitionClosingEventArgs args)
+            {
+                if (args.CancellationToken.IsCancellationRequested)
+                {
+                    return Task.CompletedTask;
+                }
+
+                if (args.Reason == ProcessingStoppedReason.OwnershipLost)
+                {
+                    ownershipLostDetected = true;
+                }
+
+                return Task.CompletedTask;
+            }
+
+            // Initialize the args with lost ownership reason.
+
+            var eventArgs = new PartitionClosingEventArgs("0", ProcessingStoppedReason.OwnershipLost);
+
+            // Execute handler then validate that partion's lost ownership has been detected.
+
+            await closeEventHandler(eventArgs);
+            Assert.IsTrue(ownershipLostDetected);
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Serves as a simulation of the host application for
+        ///   examples.
+        /// </summary>
+        ///
+        private static class Application
+        {
+            /// <summary>
+            ///   A simulated method that an application would register as an event handler.
+            /// </summary>
+            ///
+            /// <param name="eventArgs">The arguments associated with the event.</param>
+            ///
+            public static Task ProcessEventAsync(ProcessEventArgs eventArgs) => Task.CompletedTask;
+
+            /// <summary>
+            ///   A simulated method for handling an exception that occurs during
+            ///   event processing.
+            /// </summary>
+            ///
+            /// <param name="eventArgs">The arguments associated with the failed processing.</param>
+            /// <param name="exception">The exception to handle.</param>
+            ///
+            public static void HandleProcessingException(ProcessEventArgs eventArgs,
+                                                         Exception exception)
+            { }
```

Comment:
```suggestion
```

This was helpful for adding some visualization when explaining the handler use, but I think we can skip in favor of comments for this scenario.

Reply:
Hi @jsquire,

Thanks a lot for your feedback!

I have to focus on somethin else until next week. Sorry! I'll get back to it as soon as I'll get free time.

Cheers
A

Reply:
Not at all;  thanks very much for taking this on.    Thanks for the updated, but no need to apologize at all - there's no rush whatsoever.